src/Widgets/EventRow.vala

```diff
@@ -90,4 +99,10 @@ public class DateTime.EventRow : Gtk.ListBoxRow {
         var time_format = Granite.DateTime.get_default_time_format (time_manager.is_12h);
         time_label.label = "<small>%s – %s</small>".printf (cal_event.start_time.format (time_format), cal_event.end_time.format (time_format));
     }
+    public void set_color () {
+        var provider = new Gtk.CssProvider ();
+        Util.get_style_calendar_color (cal_event.cal, provider);
+        grid_context.add_provider (provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);
+        event_image_context.add_provider (provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);
```

Comment:
Not sure you should be adding new providers for color without removing the old ones. Why not keep one color provider per instance and just update it?

Reply:
It's how it works for adding color without using the screen method. :man_shrugging:

Reply:
It works OK if you have an instance member `color_css_provider` rather than creating a new one when the color changes.  However, I am not sure you need to connect to the `notify["color"]` signal, in practice.  I cannot find a way of changing the color of the calendar without the indicator losing focus and closing. If you do not connect to this signal then the provider only gets added once anyway so you do not need an instance member.

Reply:
The color signal is for e.g. when the user changes the Calendar color externally, so I'm afraid I can't remove it. I will however think about how to implement color_css_provider. A snippet of code would help me immensely.

Reply:
I still cannot think of a way of changing the calendar color externally while maintaining focus on this calendar indicator. When this indicator loses focus it closes and the color will be updated when it is re-opened. Or are you thinking that the calendar may be accessed from another computer while the indicator is open?  Pretty unlikely, but I suppose it may be possible.  I'll give a code snippet below.

Reply:
Fixed.